runtime/validation.py

```diff
@@ -77,6 +77,20 @@ def validate_descriptor_dtype(dataset: str, features_array: np.ndarray):
             f"dtype is: {features_array.dtype}"
         )
 
+def validate_descriptor_dim(dataset: str, features_array: np.ndarray, max_dim: int = 512):
+    if (submitted_dim := features_array.shape[1])  > max_dim:
+        raise DataValidationError(
+          f"Features array for {dataset} exceeds max dim of {max_dim}: "
+          f"submitted dim is {submitted_dim}."
+        )
+
+def validate_sorted_ids(dataset: str, video_ids: np.array):
+    if not np.all(video_ids[:-1] <= video_ids[1:]):
+        indices = np.argwhere(video_ids[:-1] > video_ids[1:])
```

Comment:
Might also consider:
```suggestion
    is_sorted = video_ids[:-1] <= video_ids[1:]
    if not np.all(is_sorted):
        indices = np.argwhere(~is_sorted)
```
But feel free to ignore too.

Reply:
i might walrus here too

Reply:
nah too messy this is better